### PR TITLE
Gridify Filter

### PR DESCRIPTION
**Pull request by nathanblair**
Okay, yes, the name is...silly. Disregard that (and the group/subgroup classification of the filter) for right now. This filter was created to be consumed either as a discrete filter or from some lower level code for the upcoming Blend filter. Blend and stitching algorithms would require spatial awareness and relationships of various data containers. This filter (as of now) provides an attribute matrix ("Grid") with a singular attribute array ("Layout") that houses a simple UInt16BitArrayType (unsigned short) of 1 tuple with 3 components (size of 3), of which each component describes the (column, row, depth) orientation of that particular data container relative to other image geometry data containers in the data container array. It does so by determining the relationships of the origins for each data container.

This PR is mostly aimed at discussing the code used to create this filter as a utility:
- All on it's own
- Integrated into a Geometry (ImageGeom)
- Integrated into the Blend filter / a stitching algorithm filter later
- All of the above?
- Something else entirely

There are multiple for-loops in the datacheck() method that could be severely optimized and I'm working out how best to do that (its probably more a data structure/organization problem than it is an algorithm optimization problem).

This filter could provide data useful for the upcoming blend algorithm as well as any other filters that may benefit from knowing the spatial relationship between image geometry data containers.

### CODE REVIEW

**Comment by nathanblair** [comment]
I should note that commit 64faa7e does not fix #29 of the DREAM3D repo but rather my own issue on my personal fork (nathanblair/DREAM3D#29).

Please disregard that in the commit messages listed.

File: Source/Plugins/OrientationAnalysis/Test/GridifyTest.cpp [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    //    QFile::remove(UnitTest::GridifyTest::TestFile1);
+    //    QFile::remove(UnitTest::GridifyTest::TestFile2);
+#endif
+    return 0;
+  }
+
+  int SetUp()
+  {
+    return 0;
+  }
```
Comment (mmarineBlueQuartz) [inline Source/Plugins/OrientationAnalysis/Test/GridifyTest.cpp:114]: This looks like something that either was forgotten to be implemented or does not need to be.  Consider removing it if it's unnecessary.

**Review by mmarineBlueQuartz — COMMENTED** [review]
Change your initializer_list assignments to standard = operations.  For instance, `uint16_t index{0};` would become `uint16_t index = 0;`  It might be less efficient, but it will follow the same standard as everything else.

If the only line in a unit test method is `return 0;`, remove the method.  RemoveTestFiles() methods should be fine as they still hold what had been performed even if it is commented out.  Others, like ValidateData() and SetUp() that do not do what their name implies, should be removed so that someone who reads part of the unit test does not wrongfully assume that the data has been validated or there is some setup work being done before another test.  Of course, if these methods are still under development and something will be there, ignore this point.

**Comment by nathanblair** [comment]
Should be g2g. I've mad the adjustments to the initialization of scalars and other trivial types to use the '=' operator.

**Comment by imikejackson** [comment]
This has been implemented in other commits.

**Pull request closed without merging**